### PR TITLE
Bump dependencies/CS standard/add typed properties

### PR DESCRIPTION
**Pull request by Ocramius**

### CODE REVIEW

File: tests/ProxyManagerTest/Signature/Exception/InvalidSignatureExceptionTest.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * @covers \ProxyManager\Signature\Exception\InvalidSignatureException
  * @group Coverage
  */
-class InvalidSignatureExceptionTest extends TestCase
+final class InvalidSignatureExceptionTest extends TestCase
 {
     public function testFromInvalidSignature() : void
```
Comment (malukenho) [inline tests/ProxyManagerTest/Signature/Exception/InvalidSignatureExceptionTest.php]: can be declared as static

**Comment by Ocramius** [comment]
Hmm, I think the CS checks simply choke on typed properties...

File: tests/ProxyManagerTest/Signature/Exception/MissingSignatureExceptionTest.php [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * @covers \ProxyManager\Signature\Exception\MissingSignatureException
  * @group Coverage
  */
-class MissingSignatureExceptionTest extends TestCase
+final class MissingSignatureExceptionTest extends TestCase
 {
     public function testFromMissingSignature() : void
```
[the remark this answers is not in the document]
Comment (Ocramius) [reply tests/ProxyManagerTest/Signature/Exception/MissingSignatureExceptionTest.php]: I wouldn't declare test methods as static - where did that come from?

File: tests/ProxyManagerTest/Signature/Exception/InvalidSignatureExceptionTest.php (hunk 1 above)
Comment (Ocramius) [reply tests/ProxyManagerTest/Signature/Exception/InvalidSignatureExceptionTest.php]: Same as above: test methods, unlikely to be made `static`

File: src/ProxyManager/Configuration.php [hunk 3]
```diff
@@ -135,6 +122,6 @@ public function setClassSignatureGenerator(ClassSignatureGeneratorInterface $cla
     public function getClassSignatureGenerator() : ClassSignatureGeneratorInterface
     {
         return $this->classSignatureGenerator
-            ?: new ClassSignatureGenerator($this->getSignatureGenerator());
+            ?? new ClassSignatureGenerator($this->getSignatureGenerator());
```
[the remark this answers is not in the document]
Comment (Ocramius) [reply src/ProxyManager/Configuration.php]: Good one, fixing that :-)

[the remark this answers is not in the document]
Comment (Ocramius) [reply src/ProxyManager/Configuration.php]: Good one, fixing that :-)

File: tests/ProxyManagerBench/BaselinePropertyAccessBench.php [hunk 4]
```diff
@@ -16,31 +16,16 @@
...
+    private ClassWithPrivateProperties $privateProperties;
+    private ReflectionProperty $accessPrivateProperty;
+    private ClassWithProtectedProperties $protectedProperties;
+    private ReflectionProperty $accessProtectedProperty;
+    private ClassWithPublicProperties $publicProperties;
+    private ClassWithMixedProperties $mixedProperties;
+    private ReflectionProperty $accessMixedPropertiesPrivate;
+    private ReflectionProperty $accessMixedPropertiesProtected;
 
     public function setUp() : void
```
Comment (malukenho) [inline tests/ProxyManagerBench/BaselinePropertyAccessBench.php]: `protected`

Comment (Ocramius) [reply tests/ProxyManagerBench/BaselinePropertyAccessBench.php]: This is a bench, not PHPUnit.

File: tests/ProxyManagerTest/Signature/Exception/MissingSignatureExceptionTest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (malukenho) [reply tests/ProxyManagerTest/Signature/Exception/MissingSignatureExceptionTest.php]: Well, as the method is disconnected from the object context you should use `static` IMO

File: tests/ProxyManagerBench/BaselinePropertyAccessBench.php (hunk 4 above)
Comment (malukenho) [reply tests/ProxyManagerBench/BaselinePropertyAccessBench.php]: Oh! That is right

File: tests/ProxyManagerTest/Signature/Exception/MissingSignatureExceptionTest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ocramius) [reply tests/ProxyManagerTest/Signature/Exception/MissingSignatureExceptionTest.php]: It is called as instance method from the test runner, so it should stay as instance method

**Comment by Ocramius** [comment]
Green: :ship:

**Pull request merged**